### DISCUSSION TITLE
[v4] Get collaborators on archived repositories

### DISCUSSION THREAD

**Discussion by bskidan** [community/community#24360] [Apps, API and Webhooks]
<p>Using GraphQL API, when trying to get collaborators for all repositories in my organization, for one repository that was archived I get the error: “Must have push access to view repository collaborators.”. Since the repository is private and has proprietary code I still want to see users within our organization who have access to the repository and can see the code (direct access or external collaborators, not through Team membership). Is there a way to get the list through the API?</p>

**Comment by nethgato** [comment]
<p>Hi hi <a class="mention" href="/bskidan">@bskidan</a> :wave:</p>
<p><code>Collaborators</code> are users added to the repository by an admin and this information is only available to the admin via the repository’s settings page or those with push access via the API.</p>
<p>Is it possible that <code>contributors</code> are what you might want to identify?</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-repository-contributors" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-repository-contributors" target="_blank" rel="noopener">Repositories - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Let us know if this is useful information. :bow:</p>

**Comment by bskidan** [comment]
<p>Hi <a class="mention" href="/nethgato">@nethgato</a>.</p>
<p>Thanks for replying. The link you’ve posted points to the REST API, I’m actually using GraphQL version (v4), since I need to know the detailed permissions, like “MAINTAIN”, which are only available in v4. I couldn’t find ‘contributors’ in the GraphQL API, and I’m pretty sure that ‘collaborators’ is what I’m after. Here is what I’m trying to do:</p>
<p>I want to audit all access to all of the repositories in my organization by all users (direct access, via teams, or outside collaborators) in one report. I want to see the same information as in the repository’s settings page, but in one place for all repos.</p>
<p>I use admin user to query the API, which has access to all the repositories, and I can see everything that I need except the collaborators on the archived repositories, even though my user has admin access to those repositories. I believe that the reason is that the archived repository is ‘read-only’ thus there is no push access, even though my user has admin permissions to the repo. However I do need to know users who can see my code.</p>
<p>So I guess the question is:</p>
<ol>
<li>Can the “push access” requirement be relieved?</li>
<li>Or is there another way to see (via GraphQL API) who has access to my archived repositories?</li>
</ol>

**Comment by nethgato** [comment]
<p>Hey <a class="mention" href="/bskidan">@bskidan</a> – thanks so much for restating your use case.</p>
<p>The reason I linked to v3, is because the same is not available in GraphQL.</p>
<p>Though, the archived behavior is more clearly the concern than how I originally perceived your post. Basically, I misread what you wrote and thanks again for rephrasing.</p>
<p>Let me see if I can dig around for a better answer for you! :bow:</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey again <a class="mention" href="/bskidan">@bskidan</a> … wow… it happens every now and then, but it’s happened this time. Not only is this something that GitHub has been aware of, it’s something I’ve been aware of.</p>
<p>I have no excuse as to why I didn’t catch this in my initial response, but I’m reporting it now.</p>
<p>We have an open internal issue, on this exact behavior. It is not expected, and I’ve made sure to add this post to our issue for continued tracking.</p>
<p>Right now, I have no ETA on a fix, but you should be able to leverage REST:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#collaborators" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#collaborators" target="_blank" rel="noopener">Repositories - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>…as a temporary workaround.</p>
<p>Though I certainly understand that if v4/GraphQL is the preferred option in your tooling, that it is less than helpful.</p>
<p>At the moment, this is where we are and I thank you or your patience, and for reporting this in our Community.</p>

**Comment by bskidan** [comment]
<p><a class="mention" href="/nethgato">@nethgato</a> thank for finding this out, I’ll use both APIs simultaneously until this is resolved</p>
